### PR TITLE
Adds victoriametrics package with vmagent subpackage

### PR DESCRIPTION
**Pull request by dakaneye**
Related to https://github.com/chainguard-dev/image-requests/issues/1955

### CODE REVIEW

File: victoriametrics.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+subpackages:
+  - name: ${{package.name}}-vmagent
+    description: vmagent is a tiny agent which helps you collect metrics from various sources, relabel and filter the collected metrics and store them in VictoriaMetrics or any other storage systems via Prometheus remote_write protocol or via VictoriaMetrics remote_write protocol.
+    pipeline:
+      - uses: go/build
+        with:
+          ldflags: -extldflags "-static"
+          toolchaintags: netgo,osusergo,nethttpomithttp2,musl
+          packages: github.com/VictoriaMetrics/VictoriaMetrics/app/vmagent
```
Comment (xnox) [inline victoriametrics.yaml:24]: static linking is either redundant (as binary is already static anyway) or does not achieve what you think it does (statically links glibc, and shared libraries creating dark matter in our containers unaccounted for with sbom).

some of the tags are redundant as already specified by our default pipeline.

Some look sus - what does musl tag do?

File: victoriametrics.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package:
+  name: victoriametrics
```
Comment (xnox) [inline victoriametrics.yaml:2]: If the package only ships a subpackages and nothing else, can you call this package vicrotiametrics-vmagent directly?

**Review by xnox — CHANGES_REQUESTED** [review]

File: victoriametrics.yaml (hunk 1 above)
Comment (dakaneye) [reply victoriametrics.yaml:24]: Ah that's probably a relic of the fact they're own docker image is alpine, i'll try removing. nethttp one seems like it's just a space saver for them. I can remove static linking as I was just following their build: https://github.com/VictoriaMetrics/VictoriaMetrics/blob/22975d28ae4871a4e6ede1e482b373e13523870b/deployment/docker/Makefile#L43 and it worked without it anyways for the fips side.

File: victoriametrics.yaml (hunk 2 above)
Comment (dakaneye) [reply victoriametrics.yaml:2]: @developer-guy plans to add several more subpackages in the near future (that are separate apps in the same monorepo)

**Review by xnox — APPROVED** [review]

**Pull request merged**